src/global/global_init.cc

```diff
@@ -292,7 +297,11 @@ void global_init_daemonize(CephContext *cct)
 	 << cpp_strerror(ret) << dendl;
     exit(1);
   }
-
+  
+ if (test_pid_file_in_use(g_conf) < 0) {
```

Comment:
as usual, conf-> daemonize is false,  global_init_daemonize() returns after call global_init_prefork(), but if conf->daemonize is true, then global_init_prefork()  doenn't check pid_file and directly returns, so under this situation, pid-file needs to check again.